include/oneapi/dpl/internal/distributed_ranges_impl/detail/ranges.hpp

```diff
@@ -163,11 +163,11 @@ struct segments_fn_
     }
 };
 
-} // namespace
+} // namespace __detail
 
-inline constexpr auto segments = segments_fn_{};
+inline constexpr auto segments = __detail::segments_fn_{};
 
-namespace
+namespace __local
```

Comment:
Why not also `__detail`?

Reply:
`__local::local_fn_` and `__detail::local_fn_` are different, not sure however at the moment why we need both

Reply:
I see, `__detail::local_fn_` uses `ranges::local` if available, otherwise behaves (almost) as `std::identity`.
I think the best is to make this behavior clear from the name - for example, name it as `local_or_identity`.

Note also that `std::identity` uses forwarding to preserve the value category, see https://eel.is/c++draft/func.identity. The "identity" `operator()` misses it.